stubs/pyformance/__init__.pyi

```diff
@@ -1,12 +1,13 @@
-from typing import Callable, Dict, Optional
+from collections import abc
+from typing import Optional
 
 from pyformance.meters import Gauge
 
 
 class MetricsRegistry:
     def __init__(self, clock): ...
-    def dump_metrics(self)-> Dict[str, Dict[str, float]]: ...
+    def dump_metrics(self)-> abc.Mapping[str, abc.Mapping[str, int|float|str]]: ...
```

Comment:
:exploding_head: I like it much more than Union(...) ! I noticed it is specific to python 3.10 and this sparked a question: how do we decide which version of the product uses which python version ?

Reply:
There are two parts to this:
1. afaik stubs are only for the type checker so we can cheat a little: mypy knows about `|` even if Python does not.
2. Generally we decide based on what's available on the lowest RHEL version we need to support for a product version. For both iso5 and iso6 that's RHEL 8, which doesn't include Python 3.10 yet, so we're stuck with 3.9 for now. For iso7, we might be able to upgrade to 3.10 or 3.11 eventually, if RHEL 9 includes them (I expect it will). We always have an officially supported version, and we try to be compatible with higher versions as well but we provide no such guarantees.